sdk/src/test/java/com/qonversion/android/sdk/internal/userProperties/UserPropertiesStorageImplTest.kt

```diff
@@ -0,0 +1,451 @@
+        }
+    }
+
+    @Nested
+    inner class Delete {
+        private lateinit var spykStorage: UserPropertiesStorageImpl
+
+        @BeforeEach
+        fun setUp() {
+            spykStorage = spyk(userPropertiesStorage)
+        }
+
+        @Test
+        fun `delete non-existent property`() {
+            // given
+            val key = "keyToDelete"
+
+            val existingProperties = mutableMapOf("key" to "value")
+            every {
+                spykStorage.putPropertiesToStorage()
+            } just runs
+
+            every {
+                spykStorage.properties
+            } returns existingProperties
+
+            // when
+            spykStorage.delete(key)
+
+            // then
+            verify(exactly = 1) {
+                spykStorage.putPropertiesToStorage()
+            }
+            val updatedProperties = spykStorage.properties
+            assertThat(updatedProperties).isEqualTo(existingProperties)
+        }
+
+        @Test
+        fun `delete existent property`() {
+            // given
+            val key = "keyToDelete"
+
+            val existingProperties = mutableMapOf("keyToDelete" to "value")
+            every {
+                spykStorage.putPropertiesToStorage()
+            } just runs
+
+            every {
+                spykStorage.properties
+            } returns existingProperties
+
+            // when
+            spykStorage.delete(key)
+
+            // then
+            verify(exactly = 1) {
+                spykStorage.putPropertiesToStorage()
+            }
+            val updatedProperties = spykStorage.properties
+            assertThat(updatedProperties).isEmpty()
+        }
+    }
+
+    @Nested
+    inner class DeleteProperties {
+        private lateinit var spykStorage: UserPropertiesStorageImpl
+
+        @BeforeEach
+        fun setUp() {
+            spykStorage = spyk(userPropertiesStorage)
+        }
+
+        @Test
+        fun `delete non-existent properties`() {
+            // given
+            val keys = listOf("keyToDelete1", "keyToDelete2")
+
+            val existingProperties = mutableMapOf("key" to "value")
+            every {
+                spykStorage.putPropertiesToStorage()
+            } just runs
+
+            every {
+                spykStorage.properties
+            } returns existingProperties
+
+            // when
+            spykStorage.delete(keys)
+
+            // then
+            verify(exactly = 1) {
+                spykStorage.putPropertiesToStorage()
+            }
+            val updatedProperties = spykStorage.properties
+            assertThat(updatedProperties).isEqualTo(existingProperties)
+        }
+
+        @Test
+        fun `delete existent properties`() {
+            // given
+            val keys = listOf("keyToDelete1", "keyToDelete2")
+
+            val existingProperties =
+                mutableMapOf("keyToDelete1" to "value1", "keyToDelete2" to "value2")
+
```

Comment:
I think we should add some values here because current tests for delete methods will pass even if the implementation will be just a `map.clear()` call